### PR TITLE
Fronend_V2: Show submission details in 3 columns on All submissions page

### PR DESCRIPTION
**Pull request by savish28**
Issue Fix: #3335

Before:
Screenshot from 2021-03-24 19-13-38

After:
Screenshot from 2021-03-24 19-36-23

### CODE REVIEW

**Comment by savish28** [comment]
@Ram81 Ping, PR up for review.

File: frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html [hunk 1]
```diff
@@ -180,38 +180,19 @@ <h5 class="fw-light">All Submissions</h5>
...
+                  <div class="col-sm-6 sol-md-6 col-lg-5 no-pad-lr">
                     <p>
                       <strong class="text-light-black fw-semibold fs-12">Status</strong> &nbsp;
                       <strong class="fw-semibold fs-14" [ngClass]="element.status" class="fs-12">
                         {{ element.status | uppercase }}
                       </strong>
                     </p>
                   </div>
-                  <div class="col-sm-6 sol-md-6 col-lg-8 no-pad-lr">
+                  <div class="col-sm-6 sol-md-6 col-lg-7 no-pad-lr">
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html:195]: @savish28 why are the column size inconsistent across divs? We should have equal size

File: frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html [hunk 2]
```diff
@@ -228,6 +209,26 @@ <h5 class="fw-light">All Submissions</h5>
                     </strong>
                   </p>
                 </div>
+                <div class="element">
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html:212]: please rename the css class to something relevant

File: frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html [hunk 3]
```diff
@@ -228,6 +209,26 @@ <h5 class="fw-light">All Submissions</h5>
...
+                      >Metadata file&nbsp;<i class="fa fa-file fs-12" aria-hidden="true"></i></strong
+                    >&nbsp;
+                    <a
+                      target="_blank"
+                      class="blue-text fs-12"
+                      [href]="element.submission_metadata_file"
+                      *ngIf="element.submission_metadata_file"
+                    >
+                      <i class="fa fa-external-link"></i> Link
+                    </a>
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html:224]: @savish28 we want all the options to be vertically aligned in a column. Can you please make that change too. Also, you can refer the `My submissions` page alignment for this.

File: frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.scss [hunk 4]
```diff
@@ -124,6 +136,14 @@ tr.element-row.expanded-row {
...
     margin-top: 20px;
   }
+  .elements-status {
+    border-right-style: none;
+    margin-left: 0;
+    border-bottom-style: solid;
+    border-bottom-color: lightgrey;
+    border-bottom-width: 1px;
+    padding-bottom: 20px;
+  }
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.scss:146]: add newline

File: frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.scss [hunk 5]
```diff
@@ -124,6 +136,14 @@ tr.element-row.expanded-row {
     margin-left: 0;
     margin-top: 20px;
   }
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.scss:138]: add newline

**Review by Ram81 — CHANGES_REQUESTED** [review]
@savish28 it would be good if you could work on 1 issue at one time. Another thing I want to point out is to wait for the assignment of an issue before starting work on it. We know everyone is excited about contributing we just want to make sure all contributors get a chance to work on issues and there is no redundant effort 🙂

I have reviewed the PR but please keep this in mind for future work.

File: frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html (hunk 2 above)
Comment (savish28) [reply frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html:212]: @Ram81, the name element is used across `challengviewsallsubmissions.components.html` and `challengsubmission.components.html` to represent a block with column flex property. All the class nomenclature in these files are revolving around `element` like elements, elements-detail, elements-status, and has been consistent all across. So should I change the name at all its occurrences and corresponding CSS across all the files or keep it as is?

File: frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html (hunk 3 above)
Comment (savish28) [reply frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html:224]: @Ram81 By options which one of the following arrangement you mean
- Vertically aligning all the three columns i.e with only one detail, status and action per row.
or
- Vertically aligning only the actions `leaderboard, Flagged, Re-run submission, Delete submission`

Comment (Ram81) [reply frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html:224]: @savish28 we should vertically align all fields for all 3 columns

File: frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html (hunk 2 above)
Comment (Ram81) [reply frontend_v2/src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html:212]: I see, let's not change it then

**Comment by savish28** [comment]
@Ram81 Thanks for the suggestions, after updating the review changes the UI looks like this:

Screenshot from 2021-03-26 17-57-01

**Comment by Ram81** [comment]
@savish28 the padding on left and right side is inconsistent when the tab is opened. Can we fix that

**Comment by savish28** [comment]
@Ram81 Is it better now?
Screenshot from 2021-03-29 17-46-53

**Comment by Ram81** [comment]
@savish28 can you also share a screenshot with the case where there are ~5 team members in a team.

**Comment by savish28** [comment]
> @savish28 can you also share a screenshot with the case where there are ~5 team members in a team.

@Ram81 Here it is!
Screenshot from 2021-03-30 12-00-08

**Comment by savish28** [comment]
@Ram81 After resolving conflicts from a recent commit, it looks like this:

Screenshot from 2021-03-30 20-46-59

**Comment by Ram81** [comment]
@savish28 can you also make font consistent with rest of the page. It looks like bunch of fields have font weight and style mismatch like `submission number`, `Submitted At`, `Re-run submission` and others.

**Comment by savish28** [comment]
> @savish28 can you also make font consistent with rest of the page. It looks like bunch of fields have font weight and style mismatch like `submission number`, `Submitted At`, `Re-run submission` and others.

@Ram81 Something like this?
Screenshot from 2021-03-31 20-49-03

**Review by Ram81 — CHANGES_REQUESTED** [review]
@savish28 `Re-run submission` and `Delete submission` field has inconsistent font. Can we fix that too

**Comment by savish28** [comment]
> @savish28 `Re-run submission` and `Delete submission` field has inconsistent font. Can we fix that too

@Ram81 Something like this?
Screenshot from 2021-04-04 19-06-12

**Pull request merged**